### ISSUE TITLE
delz increment is wrong sign when write_fv3_incr=T

### ISSUE THREAD

**Issue by jswhit** [NOAA-EMC/GSI#117]
Running a C96 3DVar experiment, I noticed that if I set write_fv3_incr=T and pass the resulting increment file directly to the model I get a different answer than if I set write_fv3_incr=F and calculate the increments using the calc_increment_ncio utility 
![image](https://user-images.githubusercontent.com/579593/109096022-042df180-76da-11eb-8ea4-99b487e2d6e2.png)
PR #116 changes the sign of the delz increment in write_incr.F90, by changing the way delz is calculated to be consistent with the way FV3 wants it (negative definite).  Re-running the same experiment with this change fixes this discrepancy.
![image](https://user-images.githubusercontent.com/579593/109096349-9e8e3500-76da-11eb-9a16-56bee98fcc3f.png)

PR #116 also changes the way delz is calculated in netcdfgfs_io.f90 to be consistent.  This does not change results since the sign of the increment had been reversed before adding to the background.

I suspect that this bug (if indeed is a bug) is affecting GFSv16.  I know we've wrestled with this several times before, so it's quite possible I've got this wrong.

@CatherineThomas-NOAA @CoryMartin-NOAA @dtkleist  @HenryWinterbottom-NOAA @RussTreadon-NOAA

**Comment by CatherineThomas-NOAA** [comment]
Thanks for bringing this up, Jeff.  The intention for the GSI delz calculation is to compute it in the positive sense (for consistency with the old model) and then flip the sign before writing out the increment.  As you point out, this is done in netcdfgfs_io.f90 and I remember going through this specifically for the increment as well, but I don't see where it is done in the current write_incr.F90.

Digging through the logs, I found where the sign flip was originally added in write_incr.F90 (d095b38d93b32778225616ef6f96df14c9e560e5):

```
          ! GSI is N->S, we want S->N
           do j=1,grd%nlat-2
-             grid(:,j) = gridrev(:,grd%nlat-1-j)
+             grid(:,j) = gridrev(:,grd%nlat-1-j) * -1.0_r_kind ! flip sign
           end do
```
but it looks like the increment code has had an overhaul since then and I don't see a sign flip anymore.  I'll let @CoryMartin-NOAA chime in on this in case I am missing something as well (he's currently on leave), but it's looking to me like there is another delz issue... It never ends.

**Comment by CatherineThomas-NOAA** [comment]
> it looks like the increment code has had an overhaul since then

The overhaul I referred to looks to be the parallel write.  So it seems like there were seemingly identical blocks of code for writing the increment replaced by a different set of fully identical blocks of code to write in parallel, where the sign flip got missed.

But there's one thing that I can't wrap my head around.  When we first tried to include the delz and delp increments with v15 development, we didn't know that the model wanted negative delz, so we gave the model the wrong sign increment...and the model blew up quickly.  This potential bug would have been in the real time parallel since we added the parallel write, which was about a year ago.  Why hasn't this caused any issues?  Maybe the sign flip is being accounted for somewhere else that is not obvious to me.

**Comment by CoryMartin-NOAA** [comment]
Based on a quick look, I agree I think this was a bug/oversight and Jeff's PR makes the code correct. However, I agree with Cathy, it is surprising to me that this sort of error would go undetected for a year without instability/crashes. Could L127 vs L64 account for why we aren't seeing the crashes like those found in v15? If this is wrong, @dtkleist what do we do so close to implementation? Fix or leave as is?

**Comment by dtkleist** [comment]
I need some confirmation and assessment of this before deciding how to proceed with v16 (and beyond), since we are in the 30 day IT test. We cannot change things without derailing the implementation altogether. Can someone pull together a quick test for a single case to at least document the impact on the analysis and subsequent forecast. We could pursue a post-implementation bug-fix that would include zeroing out the delz increment as a risk mitigation (consistent with current operations) or flip the sign if we are confident (but that would require quite a bit of testing). This will be difficult to navigate as we also have the pressures of getting commercial gpsro data in.

@CatherineThomas-NOAA @RussTreadon-NOAA : How about we re-run the v16 parallel on Dell3.5 for a few days to at least get a quick assessment?

**Comment by RussTreadon-NOAA** [comment]
GFS v16 is in the final 30-day IT test.  Submitting code updates at this
point will at a minimum reset the 30-day clock.  If an update changes
science, the impact on the GFS v16 implementation will be much more
significant.   Is global_enkf.x generating the correct sign for increments?

**Comment by CatherineThomas-NOAA** [comment]
A quick look indicates that it is correct in the enkf (line 3994 of gridio_gfs.f90):
        `inc3d(:,:,ki)=reshape(delzb-ug,(/nlons,nlats/))`

Rather than analysis minus background, it's background minus analysis.  But I will spend more time combing through the code to confirm.

**Comment by RussTreadon-NOAA** [comment]
Thank you, Cathy.  I set up the v16rt2c on Mars phase 3.   It is a gdas
only parallel using NCO v16 ICs to start from 2021022500.   The gdasanal
job is now running using global_gsi.x built from Jeff's branch,
fix_delz_inc.

**Comment by jswhit** [comment]
Thanks @RussTreadon-NOAA.  Will the data go to HPSS in /NCEPDEV/emc-global/5year/emc.glopara/WCOSS_D/gfsv16?

**Comment by RussTreadon-NOAA** [comment]
No, this output is not sent to HPSS.  We are currently using our v16 HPSS
bandwidth to save output from NCO's v16 parallel.   The test I am running
is not a fully cycled parallel.  We can not afford to run such a parallel
in production white space.  I mannually ran gdasanal, gdaspost*, and
gdasfcst for 2021022500. The gdasanal uses your global_gsi.x.   I am now
running the 2021022506 gdasanal using NCO's 2021022500 v16 enkf files and
the hi-res backgrounds from the test.

**Comment by RussTreadon-NOAA** [comment]
Attached are sonde o-g plots for 2021022506 from the Mars v16rt2c which uses a global_gsi.x build from fix_delz_inc.    Fits from the fix_delz_inc test generally look better than those from NCO's v16 parallel.

![gsistat_uvtq_Count](https://user-images.githubusercontent.com/26926959/109209450-8d483580-7779-11eb-9b8a-dbbf86a04bd2.png)
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/26926959/109209461-91745300-7779-11eb-9be2-8a6871edcdcb.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/26926959/109209475-95a07080-7779-11eb-8a6e-0ccf51c1e64e.png)

**Comment by RussTreadon-NOAA** [comment]
The above test has flaws.  The 2021022400 enkf forecasts were not rerun in order to save time.   This is not correct.  The 202102400 enkf increments should be recentered about the deterministic analysis.   This noted, the test results indicate an improvement from using `fix_delz_inc`.   Given this, set up v16rt2c on Mars.   The v16rt2c is configured as follows:

- start from 2021022400 gdas prep using ICs from NCO's v16 parallel
- turn off waves
- only run gfs fcst at 00Z and only out to F240
- use `global_gsi.x` built from `release/gfsda.v16.0.0` with `fix_delz_inc` changes added
- all other DA components and other GFS v16 components taken from workflow branch `release/gfsv16.0.0`

**Comment by jswhit** [comment]
I ran a longer C96 3dvar experiment with and without this fix.  Here are the innovation stats
![image](https://user-images.githubusercontent.com/579593/109322610-badfbe00-780f-11eb-960a-29b3cf5bfc0f.png)

**Comment by CatherineThomas-NOAA** [comment]
I went through the code and summarized what I found in [this presentation](https://docs.google.com/presentation/d/1ELhWSK3CRYn0qeavPIIMKVExKgEAYjluuZWUeRZVaUc/edit?usp=sharing).  There seem to be many more issues with the sign of delz throughout the code.  I traced back some of the errors and more than one is a bad merge conflict resolution.  Some appear to be oversights from rapidly adding the all the new write routines.

Can some of you go through my logic in the slides and see if it all checks out?  As we've learned too many times now, all of this sign flipping can get confusing.

**Comment by CoryMartin-NOAA** [comment]
So, @CatherineThomas-NOAA , do I understand correctly that while some places are not flipped and some are flipped twice? So it's consistent but consistently wrong?

**Comment by CatherineThomas-NOAA** [comment]
@CoryMartin-NOAA: Yes, some places are flipped twice, but I wouldn't say it's consistently wrong.  It depends which branch of the code you are going through as some places are correct (write increment vs analysis for instance).  

The nemsio double flip was almost definitely a bad merge conflict resolution.  There was an additional line of sign flipping added to the nemsio write routine with a commit only about ozone and it was an old line that was previously removed but reappeared on a later merge.

The EnKF netcdf double flips come from the calculation of delz itself when the layer indices were reversed (bottom to top then top to bottom), which flipped the sign of delz due to the log of the pressures.  But the sign flip of computing the increment as background minus analysis remained.

**Comment by jswhit** [comment]
Thanks Cathy for unravelling this mess.  I suggest that we get rid of the bug-prone sign flips entirely, and just calculate the delz as negative everywhere. We might still need the sign flip for the corner case of nemsio where the background delz is positive.

**Comment by CatherineThomas-NOAA** [comment]
I agree, Jeff.  All the delz calculations occur inside their respective nemsio/netcdf write routines anyway so its easy to separate.  Most of this sign flipping was just inherited from the old way, so no need to keep it anymore.

**Comment by jswhit** [comment]
I've updated the PR so that all calculations of delz in GSI and EnKF produce negative definite values, and all increment calculations are analysis - forecast.  The sign of the increment  is flipped only in the case of nemsio if the background delz from the nemsio file is positive.  Still need to test when hera comes back.

**Comment by jswhit** [comment]
PR updated again - noticed that the logic in gridio_gfs.f90  to compute negative delz was wrong since ak,bk are already reversed to go from bottom to top.

**Comment by jswhit** [comment]
I haven't been able to test nemsio, but I'm pretty confident the netcdf delz is correct now in both the GSI and EnKF.

**Comment by jswhit** [comment]
@dtkleist - is there anything else you need in order to make a decision on how to proceed?

**Comment by dtkleist** [comment]
@jswhit -- Given where we are, v16 is going to have to proceed as-is. As of now, I am considering a request for a bug fix immediately after the implementation or to include this as part of GFSv16.1. V16.1 is the upgrade to start actively assimilating commercial GPSRO as part of DO-2 (data from GeoOptics). The code handoff for that is end up April with the upgrade schedule for May 18.

The options are (1) set the delz increments to zero as we run things now in v15, or (2) fix the actual bug. We need to generate as much evidence as possible in a short amount of time if (2) is going to be viable. @RussTreadon-NOAA has started a full-resolution run on WCOSS to assess (2). @CoryMartin-NOAA & @CatherineThomas-NOAA have started half-resolution runs on Hera to assess both (1) and (2).

If you have the capacity, any additional experimentation with v16-like configurations would be helpful.

**Comment by jswhit** [comment]
@CatherineThomas-NOAA and @CoryMartin-NOAA - what period are you running and where is the output going on HPSS?

**Comment by CoryMartin-NOAA** [comment]
First analysis at 06z on Aug 30, 2020, and on HPSS, /NCEPDEV/emc-da/2year/Cory.R.Martin/HERA/scratch/v16x_delz_flip and /NCEPDEV/emc-da/2year/Cory.R.Martin/HERA/scratch/v16x_delz_zero

**Comment by CoryMartin-NOAA** [comment]
@CatherineThomas-NOAA has a control, not sure of location on HPSS, but archive gsistat are on Hera at /scratch1/NCEPDEV/da/Catherine.Thomas/archive/v16x_sept_ctl/

**Comment by CatherineThomas-NOAA** [comment]
@jswhit The control is on HPSS here:
/NCEPDEV/emc-da/2year/Catherine.Thomas/HERA/scratch/v16x_sept_ctl

**Comment by jswhit** [comment]
OK thanks.  Phil and I will post some diagnostics here.  I may also run a complementary hybrid gain experiment.

**Comment by jswhit** [comment]
Initial results (small sample!) show that v16x_delz_flip is perhaps slightly better than v16x_delz_zero, and more significantly better than v16x_sept_ctl.  I'll keep these plots updated as the experiments proceed.

![image](https://user-images.githubusercontent.com/579593/110256551-ba090380-7f56-11eb-8edb-3bc319abb581.png)

![image](https://user-images.githubusercontent.com/579593/110256560-c5f4c580-7f56-11eb-9886-c45edb5303b3.png)

**Comment by CoryMartin-NOAA** [comment]
![Screen Shot 2021-03-08 at 8 58 14 AM](https://user-images.githubusercontent.com/6354668/110333364-84642900-7fef-11eb-929e-2fb7283d330b.png)
![Screen Shot 2021-03-08 at 8 57 42 AM](https://user-images.githubusercontent.com/6354668/110333376-875f1980-7fef-11eb-818b-14fe4a35dd4b.png)

My experiments also have very similar results over the similar period. RMSE looks slightly better but I feel like the bias is noticeably reduced.

**Comment by dtkleist** [comment]
From @CoryMartin-NOAA plots, that is a pretty significant change in temperature bias between 300-100 hPa. Are we also able to make better use of satellite data in this region (ATMS Ch. 8-9 for example) or is the bias correction too slow to adjust?

**Comment by CoryMartin-NOAA** [comment]
@dtkleist later today I'll try to whip up a quick plot of mean ATMS 8-9 O-F stats. Theoretically this should be fairly straightforward with the PyGSI tools we implemented from @aerorahul that parse the gsistat files.

**Comment by RussTreadon-NOAA** [comment]
Attached are sonde fits plots from the v16rt2c (delz fix) and NCO's v16
parallel (v16rt2n)

**Comment by RussTreadon-NOAA** [comment]
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/26926959/110336750-2c2f2600-7ff3-11eb-9cb5-939366baaa80.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/26926959/110336770-32250700-7ff3-11eb-8b83-e09718de5167.png)

A bit more of a mixed bag at operational resolution.

**Comment by jswhit2** [comment]
atms_npp channel 8  background fits are better for almost every analysis time

![image](https://user-images.githubusercontent.com/7596057/110357309-63540600-7ff8-11eb-89cf-25c22f71e93e.png)

**Comment by jswhit2** [comment]
same for atms_n20 channel 8

![image](https://user-images.githubusercontent.com/7596057/110357965-16bcfa80-7ff9-11eb-9048-cbb5057521c2.png)

**Comment by RussTreadon-NOAA** [comment]
Attached are time series of ATMS NPP and N20 channel 8 rms and bias (qc'd data after bias correction) from the v16rt2n (NCO v16) and v16rt2c (delz fix).   The time period plotted is 2021030600 through 2021030818.  The v16rt2c rms is reduced with respect to v16rt2n.   The v16rt2c bias is lower.   For some cycles "_lower_" means a larger bias in an absolute sense.

![atms8_npp_rms](https://user-images.githubusercontent.com/26926959/110507652-32de9b80-80ce-11eb-8f25-8c3857464e92.png)
![atms8_npp_bias](https://user-images.githubusercontent.com/26926959/110507665-36722280-80ce-11eb-8155-7e90606a7e10.png)
![atms8_n20_rms](https://user-images.githubusercontent.com/26926959/110507677-396d1300-80ce-11eb-9445-1dadb7986495.png)
![atms8_n20_bias](https://user-images.githubusercontent.com/26926959/110507684-3c680380-80ce-11eb-8ec8-2b0a37bef6f7.png)

**Comment by CoryMartin-NOAA** [comment]
![Screen Shot 2021-03-09 at 1 34 54 PM](https://user-images.githubusercontent.com/6354668/110520417-6e806200-80dc-11eb-81ed-3acd08af884f.png)
![Screen Shot 2021-03-09 at 1 34 00 PM](https://user-images.githubusercontent.com/6354668/110520420-704a2580-80dc-11eb-8534-eb4e10b51e19.png)
5 more cycles in these plots, results look similar from before, only thing I notice is perhaps a greater reduction in bias for winds in the upper troposphere

**Comment by jswhit** [comment]
I'm running a hybrid gain experiment with my own workflow and it seems to still be performing better.  Still need to track down why - they should be basically identical.

![image](https://user-images.githubusercontent.com/579593/110570780-b8d10580-8113-11eb-9242-2cb30bc840e6.png)

**Comment by jswhit** [comment]
It looks like most, if not all, of this difference is coming from the the use of the ensemble mean.  Here's a plot of the innovation stats from v16x_delz_flip for both the high-res control forecast (from gdas.tar) and the ensemble mean forecast (from enkfgdas.tar)
![image](https://user-images.githubusercontent.com/579593/110660188-890c1700-8180-11eb-9923-18335c40b843.png)
The hybrid gain control forecast in my experiment is generated by 'replaying' to the ensemble mean, so it looks more like the ensemble mean forecast.  Question is, why does the ensemble mean forecast fit the obs so much better than the high-res control?

**Comment by jswhit** [comment]
For the sake of completeness, here's a plot comparing the innovation stats for the ensemble means in both experiments.
![image](https://user-images.githubusercontent.com/579593/110663100-616a7e00-8183-11eb-8b9b-a52da3b843f1.png)
So the difference is mainly coming from the high res control forecast.

**Comment by CoryMartin-NOAA** [comment]
Update:
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/111174092-8a24b600-857d-11eb-8679-805715d63e35.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/111174093-8a24b600-857d-11eb-9e48-a1121c8fe466.png)

**Comment by RussTreadon-NOAA** [comment]
Update from Mars v16rt2c.  v16rt2n is NCO's gfs v16 parallel.

![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/26926959/111232390-68e5b900-85c1-11eb-9f6f-7ff4afb67ef3.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/26926959/111232396-6c794000-85c1-11eb-85b8-38630909d717.png)

**Comment by CoryMartin-NOAA** [comment]
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/111784357-b6984500-8891-11eb-807d-8bff8b4c5ee3.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/111784358-b6984500-8891-11eb-9490-9f9c21e82e48.png)
Update with 2 weeks of cycles

**Comment by CoryMartin-NOAA** [comment]
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/111793660-db91b580-889b-11eb-8016-a7cebf4ce153.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/111793663-db91b580-889b-11eb-8e22-1cfe9513c533.png)

**Comment by jswhit** [comment]
same thing as Cory but will all in-situ obs and statistical significance shaded


![image](https://user-images.githubusercontent.com/579593/111805968-1cd39680-8897-11eb-9bd9-bfc129512bb1.png)

![image](https://user-images.githubusercontent.com/579593/111806233-66bc7c80-8897-11eb-960a-73f84aa3ecf7.png)

**Comment by jswhit** [comment]
and here is the difference between the control forecast and the ensemble mean forecast for v16x_delz_flip

![image](https://user-images.githubusercontent.com/579593/111806802-f7935800-8897-11eb-9c45-6269eee110d0.png)

**Comment by CoryMartin-NOAA** [comment]
Three more days worth of cycles
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/112027353-79d88200-8b0d-11eb-98ca-d9ee559bd984.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/112027356-79d88200-8b0d-11eb-82d1-7a643927858c.png)

**Comment by CoryMartin-NOAA** [comment]
Now with over 3 weeks of cycles.

![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/112847804-6e3d0c00-9075-11eb-8ecb-6ff2a6fd3bca.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/112847805-6e3d0c00-9075-11eb-8e00-0c459326fbba.png)

**Comment by RussTreadon-NOAA** [comment]
**v16rt2c update**

Mars v16rt2c vsdb and metplus pages are now available.   As a reminder, the v16rt2c was warm started from NCO's v16 parallel on 2021030200.   The v16rt2c only runs the gfs cycle at 00Z.  The gfs forecast is run out to F240 (10 days).  So far the v16rt2c is able to keep up with real time using Mars dev2.   Hopefully this throughput level can be maintained into April.

NCO implemented v16 at 2021032212.   Two v16rt2c pages with monitoring and verification plots have been created - one before and one after 2021032212 - at the links provided below:

- [13 to 22 March 2021](https://www.emc.ncep.noaa.gov/gmb/emc.glopara/vsdb/v16rt2c_13_22mar/)
- [23 March to present](https://www.emc.ncep.noaa.gov/gmb/emc.glopara/vsdb/v16rt2c/)

Each page contains links to vsdb, fit2obs, and metplus plots.   A daily cron kicks off plotting jobs at 1330Z to update the 3/23-present page.   Most plot jobs finish daily by 17Z.

Care should be exercised when interpreting GFS (operational) verification on the 3/23-present page.   Currently the GFS forecasts are a mix of v15 and v16 GFS forecasts - all verified using GFS v16 analyses.   As the v16rt2c cycles further into April more of the GFS verification will be GFS v16 forecasts verified using GFS v16 analyses.    At some point we may want to reset the start date for 3/23-present verification so that all GFS forecasts originated from v16 initial conditions.

**Comment by jswhit** [comment]
Looks like a solid improvement - even though it's a small sample size

**Comment by CoryMartin-NOAA** [comment]
Another update now with 4 weeks of cycles

![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/113421467-27af2080-9399-11eb-9a8a-30e4fb089d70.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/113421468-27af2080-9399-11eb-93b7-01262e883a1b.png)

**Comment by RussTreadon-NOAA** [comment]
**Operational GFS v16 -vs- v16rt2c**

PyGSI profiles for sonde (o-f) bias and rmse averaged over the period 2021032212 (GFS v16 implementation) and 2021040200.   Plotted data taken from gdas gsistat files.

![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/26926959/113426530-9a23fe80-93a1-11eb-9172-5590eca3d0a6.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/26926959/113426536-9d1eef00-93a1-11eb-9eab-c8667d22f3ec.png)

**Comment by CoryMartin-NOAA** [comment]
![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/114421422-496f8b00-9b83-11eb-8f80-1dbf52a59b4f.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/114421425-496f8b00-9b83-11eb-9ff4-fc77ca748d5d.png)
Update for another week or so of cycles, now through Oct 06 00z

**Comment by jswhit** [comment]
Has a decision been made on whether to include this in GFSv16.1?

**Comment by RussTreadon-NOAA** [comment]
Effective 2021040606 gdas the Mars v16rt2c began assimilating commercial gprso data.   The updated [v16rt2c monitoring page](https://www.emc.ncep.noaa.gov/gmb/emc.glopara/vsdb/v16rt2c/) now includes v161rt1 for comparison.   Both [vsdb](https://www.emc.ncep.noaa.gov/gmb/emc.glopara/vsdb/v16rt2c/) and [metplus](https://www.emc.ncep.noaa.gov/gmb/emc.glopara/metplus/v16rt2c/index.html) based plots are available.   The v161rt1 and v16rt2c are built from the same NOAA-EMC/GSI branch, [release/gfsda.v16.1.0](https://github.com/NOAA-EMC/GSI/tree/release/gfsda.v16.1.0), with the v16rt2c build including local modifications to the following source code files
```
        modified:   src/enkf/gridio_gfs.f90
        modified:   src/gsi/ncepnems_io.f90
        modified:   src/gsi/netcdfgfs_io.f90
        modified:   src/gsi/write_incr.f90
```
At present both the v161rt1 and v16rt2 `$HOMEgfs/sorc/gsi.fd` are at 532fa23.

**Comment by dtkleist** [comment]
@jswhit Per Russ's response, the plan right now is to include it in v16.1 pending continued positive results from v16rt2c real time run. Final decision will be made prior to code handoff to NCO at the end of April.

**Comment by CoryMartin-NOAA** [comment]
Here are the final figures from the C384 tests on Hera. From Aug 30 - Oct 10.

![gsistat_uvtq_Bias](https://user-images.githubusercontent.com/6354668/114715141-c2482180-9d00-11eb-8612-5501e45b2adc.png)
![gsistat_uvtq_RMSE](https://user-images.githubusercontent.com/6354668/114715142-c2482180-9d00-11eb-9b93-ec9f89ddadc9.png)

**Comment by CatherineThomas-NOAA** [comment]
Attached are some figures from both sets of experiments.  At Daryl's recommendation, I tried to show whether the wrong signed delz increment created more imbalance by plotting time series of the surface pressure OmF.  The left plots with the time series overlaid don't show much of a signal to my eyes, but the difference plots favor positive values, meaning the bugfix reduces the OmF and pointing to improved balance.

[Delz_ Ps OmF.pdf](https://github.com/NOAA-EMC/GSI/files/6361303/Delz_.Ps.OmF.pdf)

**Comment by dtkleist** [comment]
Thanks @CatherineThomas-NOAA. This further strengthens the case and we will proceed to include the bug fix as part of v16.1.

**Comment by MichaelLueken-NOAA** [comment]
These changes have been merged to the authoritative repo at 51a6564. Closing this issue.
